### PR TITLE
feat(island-ui): UploadedFile component update

### PR DESCRIPTION
**Pull request by oddsson**
# UploadedFile component update

https://app.asana.com/0/1199153462262248/1200905978520084

## What

I added three things to the UploadedFile component

1. You can now set the default background color on the uploaded files.

2. You can select a “done” icon, i.e. the icon in the uploaded files component that’s shown on upload success.

3. You can now have a “Open file” handler in the uploaded files component.

## Why

The Judicial-system application uses the InputFileUpload component to upload files to an external system. Users should be able to view the files they upload and a “Open” button helps with clarity. Users can’t remove the files so it doesn’t make sense to display an x, indicating that you can. Instead, we want to display a checkmark icon, indicating that the upload was successful.

### With a "open file" handler

https://user-images.githubusercontent.com/3789875/133249281-d1b78f48-24da-4f0d-b7f3-3f3e416fffe1.mov

### Without a "open file" handler

https://user-images.githubusercontent.com/3789875/133249294-0a9ecf55-df64-488c-b5b5-91b1ffe6168e.mov

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by barabrian — COMMENTED** [review]
LGTM, just a minor comment

**Comment by oddsson** [comment]
Thanks for the review @barabrian 🙏  I appreciate it!

I've made a few changes after receiving feedback from the designers group. Would you mind taking another look?

File: libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx [hunk 1]
```diff
@@ -113,50 +120,67 @@ const UploadedFile = ({
       width="full"
       position="relative"
       title={file.name}
-      className={styles.uploadedFile}
-      onClick={(e) => e.stopPropagation()}
+      aria-labelledBy={onOpenFile ? `Opna ${file.name}` : undefined}
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx:123]: should this be translatable?

File: libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx [hunk 2]
```diff
@@ -113,50 +120,67 @@ const UploadedFile = ({
...
-        >
-          <Icon color="blue400" icon={statusIcon(file.status)} />
-        </button>
-      )}
-
+      <Box display="flex">
+        {isUploading ? (
+          <div
+            className={styles.progressIconAnimation}
+            aria-label="Hleð upp skrá"
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx:152]: should this be translatable?

File: libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx [hunk 3]
```diff
@@ -113,50 +120,67 @@ const UploadedFile = ({
...
+        ) : (
+          <button
+            type={'button'}
+            onClick={(e) => {
+              e.stopPropagation()
+              if (!isUploading) {
+                onRemoveClick(file)
+              }
+            }}
+            aria-label="Fjarlægja skrá"
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx:178]: should this be translatable?

File: libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx [hunk 4]
```diff
@@ -113,50 +120,67 @@ const UploadedFile = ({
...
+        ) : file.status === 'error' && onRetryClick ? (
+          <button
+            type={'button'}
+            onClick={(e) => {
+              e.stopPropagation()
+              if (!isUploading) {
+                onRetryClick(file)
+              }
+            }}
+            aria-label="Reyna aftur"
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx:165]: should this be translatable?

**Review by barabrian — APPROVED** [review]
LGTM! added some comments regarding text in Icelandic but I do not see it as a stopper for this PR

**Pull request merged**